src/overlays/actors/ovl_En_Ge1/z_en_ge1.c

```diff
@@ -26,95 +73,744 @@ const ActorInit En_Ge1_InitVars = {
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ge1/func_80A31BDC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ge1/func_80A31BE8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ge1/func_80A31D88.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ge1/func_80A31DE4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ge1/func_80A31E2C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ge1/func_80A31F9C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ge1/func_80A31FE0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ge1/func_80A32078.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ge1/func_80A32190.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ge1/EnGe1_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ge1/func_80A323B0.s")
+static ColliderCylinderInit sCylinderInit = {
+    { COLTYPE_UNK10, 0x00, 0x11, 0x39, 0x10, COLSHAPE_CYLINDER },
+    { 0x00, { 0x00000000, 0x00, 0x00 }, { 0x00000702, 0x00, 0x00 }, 0x00, 0x01, 0x01 },
+    { 20, 40, 0, { 0, 0, 0 } },
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ge1/func_80A323EC.s")
+static Gfx* sHairstyleDLists[] = {
+    0x06009198, // Bob
+    0x06009430, // Straight with fringe
+    0x06009690, // Spikey
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ge1/func_80A32444.s")
+static Vec3f D_80A327A8 = { 600.0f, 700.0f, 0.0f };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ge1/func_80A32598.s")
+static u64* sEyeTextures[] = {
+    0x06000708, // Mostly Open
+    0x06000F08, // Half
+    0x06001708, // Closed
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ge1/EnGe1_Draw.s")
+extern AnimationHeader D_06000228; // Idle, arms crossed
+extern FlexSkeletonHeader D_06000330;
+extern AnimationHeader D_0600A048; // Clap
+extern AnimationHeader D_0600A498; // Dismissive gesture
+
+void EnGe1_Init(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    EnGe1* this = THIS;
+
+    ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawFunc_Circle, 30.0f);
+    SkelAnime_InitFlex(globalCtx, &this->skelAnime, &D_06000330, &D_06000228, this->jointTable, this->morphTable, 16);
+    Animation_PlayOnce(&this->skelAnime, &D_06000228);
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Collider_SetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+    this->actor.colChkInfo.mass = 0xFF;
+    this->animation = &D_06000228;
+    this->animFunc = EnGe1_CueUpAnimation;
+    this->actor.unk_1F = 6;
+    Actor_SetScale(&this->actor, 0.01f);
+
+    // In Gerudo Valley
+    if (globalCtx->sceneNum == SCENE_SPOT09) {
+        this->actor.uncullZoneForward = 1000.0f;
+    } else {
+        this->actor.uncullZoneForward = 1200.0f;
+    }
```

Comment:
```suggestion
    this->actor.uncullZoneForward = (globalCtx->sceneNum == SCENE_SPOT09) ? 1000.0f : 1200.0f;
```
Ternary possible here?